apollo-runtime/src/appleMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt

```diff
@@ -1,79 +1,29 @@
 package com.apollographql.apollo3.internal
 
-import kotlinx.coroutines.CancellableContinuation
 import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.Delay
-import kotlinx.coroutines.DisposableHandle
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.InternalCoroutinesApi
-import kotlinx.coroutines.Runnable
-import platform.Foundation.NSThread
-import platform.darwin.DISPATCH_TIME_NOW
-import platform.darwin.dispatch_after
-import platform.darwin.dispatch_async
-import platform.darwin.dispatch_get_main_queue
-import platform.darwin.dispatch_time
-import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.newFixedThreadPoolContext
+import kotlinx.coroutines.newSingleThreadContext
 
-internal actual fun defaultDispatcher(requested: CoroutineDispatcher?): CoroutineDispatcher {
-  check(requested == null || requested is DefaultDispatcher) {
-    "Changing the dispatcher is not supported on Apple targets"
-  }
-  check(NSThread.isMainThread) {
-    "defaultDispatcher must be called from the main thread"
-  }
+private val defaultDispatcher = newFixedThreadPoolContext(nThreads = 32, name = "Apollo Default Dispatcher")
 
-  return DefaultDispatcher
+internal actual fun defaultDispatcher(requested: CoroutineDispatcher?): CoroutineDispatcher {
+  return requested ?: defaultDispatcher
```

Comment:
It's a bit weird to have the `requested` parameter now. Can you move the logic at the callsite?

```
val actualDispatcher = dispatcher ?: defaultDispatcher()
```